recipes/paho-mqtt-c/all/patches/0005-fix-cmake-static.patch

```diff
@@ -0,0 +1,40 @@
+diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
+index 52cea7c..f798247 100644
+--- a/src/CMakeLists.txt
++++ b/src/CMakeLists.txt
+@@ -177,7 +177,7 @@ IF (NOT PAHO_WITH_SSL)
+             TARGET_LINK_LIBRARIES(paho-mqtt3a-static ${LIBS_SYSTEM})
+             
+             IF (NOT WIN32)
+-                SET_TARGET_PROPERTIES(paho-mqtt3a-static PROPERTIES OUTPUT_NAME paho-mqtt3a)
++                SET_TARGET_PROPERTIES(paho-mqtt3a-static PROPERTIES OUTPUT_NAME paho-mqtt3a-static)
```

Comment:
Instead of a patch, it's also perfectly fine to only add the `-static` suffix in `package_info` when not on Windows.
e.g.
```python
        if not self.options.shared and self.settings.os == "Windows":
            target += "-static"
```

Reply:
Is this suppose to change the generated library name? There is already a _cmake_target defined but it doesn't appear to change the generated library name. I also don't know if -static needs to be added for a windows build. The _cmake_target hints that does need -static for windows.

```
self.cpp_info.components["_paho-mqtt-c"].names["cmake_find_package"] = self._cmake_target
self.cpp_info.components["_paho-mqtt-c"].names["cmake_find_package_multi"] = self._cmake_target

def _cmake_target(self):
       target = "paho-mqtt3"
       target += "a" if self.options.asynchronous else "c"
       if self.options.ssl:
           target += "s"
       if not self.options.shared:
           target += "-static"
       return target
 ```

Reply:
[`OUTPUT_NAME`](https://cmake.org/cmake/help/latest/prop_tgt/OUTPUT_NAME.html) sets the name of the output file.

You can do something as follows (untested):
```python
def _cmake_target(self):
       target = "paho-mqtt3"
       target += "a" if self.options.asynchronous else "c"
       if self.options.ssl:
           target += "s"
       if self.settings.os != "Windows" and not self.options.shared:
           target += "-static"
       return target
```